how-to/workspace-platform-starter/client/src/framework/platform/interopbroker.ts

```diff
@@ -461,26 +499,101 @@ export function interopOverride(
 				fdc3OpenOptions.context
 			);
 			try {
+				const isOpenByIntent = this._openOptions?.openStrategy === "INTENT";
+				let appId: string | undefined;
+				let instanceId: string | undefined;
+
 				const requestedApp = await getApp(requestedId);
 				if (isEmpty(requestedApp)) {
 					throw new Error(OpenError.AppNotFound);
 				}
-				const result = await this.launchAppWithIntent(requestedApp, openAppIntent);
-				let appId: string;
-				let instanceId: string | undefined;
-				if (typeof result.source === "string") {
-					appId = result.source;
+
+				if (isOpenByIntent) {
+					const result = await this.launchAppWithIntent(requestedApp, openAppIntent);
+					if (typeof result.source === "string") {
```

Comment:
We have utils isString for this purpose

Reply:
done and there was another typeof string.